ops.h

```diff
@@ -693,15 +693,18 @@ template <size_t k>
 static HWY_NOINLINE HWY_MAYBE_UNUSED std::discrete_distribution<int>
 create_distribution(std::array<float, k>& top_k, float temperature) {
   // re-normalize distribution
-  for (size_t i = 0; i < k; ++i) {
-    top_k[i] = exp(log(top_k[i]) / temperature);
-  }
-  float denominator = 0.0f;
-  for (size_t i = 0; i < k; ++i) {
-    denominator += top_k[i];
-  }
-  denominator = 1.0f / denominator;
-  MulByConst(denominator, top_k.data(), k);
+  namespace hn = hwy::HWY_NAMESPACE;
+  using D = hn::ScalableTag<float>;
+  const D d;
+
+  const auto temperature_inv =
+      hn::Div(hn::Set(d, 1.0f), hn::Set(d, temperature));
+
+  hn::Transform(d, top_k.data(), top_k.size(),
+                [&temperature_inv](D d, hn::Vec<D> v) HWY_ATTR {
+                  return hn::Mul(hn::Exp(d, hn::Log(d, v)), temperature_inv);
```

Comment:
Looks like we have just v/temp here - I think we want log/temp, then only exp.
We are also missing the second half of the original code?

Reply:
I am confused about vtemp things, could you please explain a bit more?

About the second half, I think it already exists inside the std::discrete_distribution.
I mean it will divide each weight by the sum of all n weights. am I missing something ?

Reply:
Sure! What I see now is exp(log(v)), which is just v, then we mul by 1/temp.
I think we intended exp(log*inv).

For the second half, I am not familiar with discrete_distribution. As a sanity check, how about we temporarily run both the old and the new code, and ensure their results are the same (which is generally a helpful methodology for porting to SIMD :) ), then we can remove the old before merging?

Reply:
Ah, `discrete_distribution` does make the second half of the original code redundant -- we don't need to divide by the sum of weights.

Thanks!